common/authentication/aws/static.go

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	awsv2 "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	v2creds "github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	"github.com/dapr/kit/logger"
+)
+
+type StaticAuth struct {
+	mu     sync.RWMutex
+	Logger logger.Logger
+
+	Region       string
+	Endpoint     *string
+	AccessKey    *string
+	SecretKey    *string
+	SessionToken *string
+
+	Clients *Clients
+	Session *session.Session
+	Cfg     *aws.Config
```

Comment:
Please, do not export these fields.

Reply:
I have to leave the Clients exported so I can mock them in unit tests. The rest I unexported.

Reply:
Please create a fake AWS client implementation for mocks. We shouldn't export this.